idaes_examples/notebooks/docs/unit_models/operations/tests/test_eg_h2o_ideal.py

```diff
@@ -0,0 +1,398 @@
+        for i in model.params._phase_component_set:
+            assert i in [
+                ("Liq", "water"),
+                ("Liq", "ethylene_glycol"),
+            ]
+
+        assert model.params.config.state_definition == FpcTP
+
+        assertStructuredAlmostEqual(
+            model.params.config.state_bounds,
+            {
+                "flow_mol_phase_comp": (0, 100, 1000, pyunits.mol / pyunits.s),
+                "temperature": (273.15, 298.15, 450, pyunits.K),
+                "pressure": (5e4, 1e5, 1e6, pyunits.Pa),
+            },
+            item_callback=_as_quantity,
+        )
+
+        assert model.params.pressure_ref.value == 1e5
+        assert model.params.temperature_ref.value == 298.15
+
+        assert model.params.water.mw.value == 18.015e-3
+        assert model.params.water.pressure_crit.value == 221.2e5
+        assert model.params.water.temperature_crit.value == 647.3
+
+        assert model.params.ethylene_glycol.mw.value == 62.069e-3
+        assert model.params.ethylene_glycol.pressure_crit.value == 77e5
+        assert model.params.ethylene_glycol.temperature_crit.value == 645
+
+        assert_units_consistent(model)
+
+
+class TestStateBlock(object):
+    @pytest.fixture(scope="class")
+    def model(self):
+        model = ConcreteModel()
+        model.params = GenericParameterBlock(**config_dict)
+
+        model.props = model.params.build_state_block([1], defined_state=True)
+
+        model.props[1].calculate_scaling_factors()
+
+        # Fix state
+        model.props[1].flow_mol_phase_comp["Liq", "water"].fix(100)
+        model.props[1].flow_mol_phase_comp["Liq", "ethylene_glycol"].fix(100)
+        model.props[1].temperature.fix(300)
+        model.props[1].pressure.fix(101325)
+
+        return model
+
+    @pytest.mark.unit
+    def test_build(self, model):
+        # Check state variable values and bounds
+        assert isinstance(model.props[1].flow_mol_phase_comp, Var)
+        assert value(model.props[1].flow_mol_phase_comp["Liq", "water"]) == 100
+        assert value(model.props[1].flow_mol_phase_comp["Liq", "ethylene_glycol"]) == 100
+        assert model.props[1].flow_mol_phase_comp["Liq", "water"].ub == 1000
+        assert model.props[1].flow_mol_phase_comp["Liq", "ethylene_glycol"].ub == 1000
+        assert model.props[1].flow_mol_phase_comp["Liq", "water"].lb == 0
+        assert model.props[1].flow_mol_phase_comp["Liq", "ethylene_glycol"].lb == 0
+
+        assert isinstance(model.props[1].pressure, Var)
+        assert value(model.props[1].pressure) == 101325
+        assert model.props[1].pressure.ub == 1e6
+        assert model.props[1].pressure.lb == 5e4
+
+        assert isinstance(model.props[1].temperature, Var)
+        assert value(model.props[1].temperature) == 300
+        assert model.props[1].temperature.ub == 450
+        assert model.props[1].temperature.lb == 273.15
+
+        assert_units_consistent(model)
```

Comment:
I am recommending we use use the DiagnsoticsToolbox and the `assert_no_structural_warnings` and `assert_no_numerical_warnings` methods in new tests. These cover a lot of the common things (DoF, unit consistency, etc.) but also check for more things we should be aware of.